internal/system/food.go

```diff
@@ -0,0 +1,88 @@
+package system
+
+import (
+	"fmt"
+
+	"github.com/arsham/neuragene/internal/asset"
+	"github.com/arsham/neuragene/internal/component"
+	"github.com/arsham/neuragene/internal/entity"
+	"github.com/arsham/neuragene/internal/geom"
+	"github.com/hajimehoshi/ebiten/v2"
+)
+
+// Food system handles the reward that entities get from interaction with food.
+type Food struct {
+	entities   *entity.Manager
+	components *component.Manager
+	assets     *asset.Manager
+	sprite     *ebiten.Image
+}
+
+var _ System = (*Food)(nil)
+
+func (a *Food) String() string { return "Food" }
+
+// setup returns an error if the entity manager or the asset manager is nil.
+func (a *Food) setup(c controller) error {
+	a.entities = c.EntityManager()
+	a.assets = c.AssetManager()
+	a.components = c.ComponentManager()
+	fmt.Print("FRUITY")
+	if a.entities == nil {
+		return fmt.Errorf("%w: entity manager", ErrInvalidArgument)
+	}
+	if a.assets == nil {
+		return fmt.Errorf("%w: asset manager", ErrInvalidArgument)
+	}
+	if a.components == nil {
+		return fmt.Errorf("%w: component manager", ErrInvalidArgument)
+	}
+	a.sprite = a.assets.Sprites()[asset.FruitApple]
+	return nil
+}
+
+// update spawns an ant every 100 frames.
+func (a *Food) update(state component.State) error {
+	if !all(state, component.StateSpawnAnts, component.StateRunning) {
```

Comment:
It should check if the StateSpawnFood is set.

Reply:
yes, definitely!